specifyweb/frontend/js_src/lib/components/SchemaViewer/TableList.tsx

```diff
@@ -42,46 +42,90 @@ export function SchemaViewerTableList<
       }),
     [sortConfig, unsortedData, applySortConfig]
   );
+  const headerElements = React.useMemo(
+    () =>
+      Object.entries(headers).map(([name, label]) => (
+        <Button.LikeLink onClick={(): void => handleSort(name as FIELD_NAME)}>
+          {label}
+          <SortIndicator fieldName={name} sortConfig={sortConfig} />
+        </Button.LikeLink>
+      )),
+    [headers, handleSort, headerClassName]
+  );
+  return (
+    <GenericSortedDataViewer<DATA>
+      cellClassName={() =>
+        'border border-gray-400 p-2 dark:border-neutral-500 print:p-1'
+      }
+      className="w-fit border border-gray-400 dark:border-neutral-500"
+      data={data}
+      getLink={getLink}
+      headerClassName="border"
+      headerElements={headerElements}
+      headers={headers}
+    />
+  );
+}
+
+export function GenericSortedDataViewer<
+  DATA extends SchemaViewerRow<RR<string, SchemaViewerValue>>
+>({
+  headers,
+  data,
+  getLink,
+  headerElements,
+  className = '',
+  headerClassName = '',
+  cellClassName,
+}: {
+  readonly headers: RR<string, JSX.Element | LocalizedString>;
+  readonly headerElements: RA<JSX.Element>;
+  readonly headerClassName?: string;
+  readonly className?: string;
+  readonly data: RA<DATA>;
+  readonly getLink: ((row: DATA) => string) | undefined;
+  readonly cellClassName?: (
+    row: DATA,
+    column: keyof DATA,
+    index: number
+  ) => string;
+}): JSX.Element {
+  const indexColumn = Object.keys(headers)[0];
+
   return (
     <div
-      className={`
-        grid-table
-        w-fit flex-1 grid-cols-[repeat(var(--cols),auto)] rounded border border-gray-400 dark:border-neutral-500 print:p-1
-        ${className}
-      `}
+      className={`grid-table flex-1 grid-cols-[repeat(var(--cols),auto)] rounded print:p-1 ${className}`}
       role="table"
       style={{ '--cols': Object.keys(headers).length } as React.CSSProperties}
     >
       <div role="row">
-        {Object.entries(headers).map(([name, label]) => (
+        {headerElements.map((element, index) => (
           <div
             className={`
-              sticky top-0 border border-gray-400 bg-[color:var(--background)]
+              sticky top-0 border-gray-400 bg-[color:var(--background)]
               p-2 font-bold dark:border-neutral-500 print:p-1
-            `}
-            key={name}
+            ${headerClassName}`}
+            key={index}
```

Comment:
if you decide to revert the type of headerElements back to `RR<FIELD_NAME, LocalizedString>`, could change this key back to `name`